perf_bench_test.go

```diff
@@ -24,127 +24,277 @@ package main
 
 import (
 	"context"
+	"encoding/csv"
 	"flag"
+	"os"
 	"strconv"
+	"strings"
 	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
+	"github.com/ease-lab/vhive/profile"
 	log "github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/require"
 )
 
 var (
-	isColdStart     = flag.Bool("coldStart", false, "Profile cold starts (default is false)")
 	vmNum           = flag.Int("vm", 2, "The number of VMs")
-	targetReqPerSec = flag.Int("requestPerSec", 2, "The target number of requests per second")
-	executionTime   = flag.Int("executionTime", 2, "The execution time of the benchmark in seconds")
+	targetReqPerSec = flag.Int("requestPerSec", 10, "The target number of requests per second")
+	executionTime   = flag.Int("executionTime", 1, "The execution time of the benchmark in seconds")
+	funcNames       = flag.String("funcNames", "helloworld", "Name of the functions to benchmark")
+
+	startVMNum = flag.Int("startVM", 4, "The start VM number for multi-VMs requests per second benchmark")
+	endVMNum   = flag.Int("endVM", 100, "The end VM number for multi-VMs requests per second benchmark")
+
+	perfExecTime = flag.Float64("perfExecTime", 0, "The execution time of perf command in seconds (run `perf stat` if bigger than 0)")
+	perfInterval = flag.Uint64("perfInterval", 100, "Print count deltas every N milliseconds")
+	perfEvents   = flag.String("perfEvents", "", "Perf events")
+	perfMetrics  = flag.String("perfMetrics", "", "Perf metrics")
 )
 
+const (
+	averageExecutionTime  = "average-execution-time"
+	realRequestsPerSecond = "real-requests-per-second"
+)
+
+func TestBenchMultiVMRequestPerSecond(t *testing.T) {
+	log.SetLevel(log.InfoLevel)
+
+	*funcNames = *funcName
+
+	var (
+		servedTh      uint64
+		pinnedFuncNum int
+		startVMID     int
+		isSyncOffload bool = true
+		metrFile           = "benchRPS.csv"
```

Comment:
if used in a single place, better to inline the constant

Reply:
Done